casper/src/main/scala/coop/rchain/casper/blocks/proposer/Proposer.scala

```diff
@@ -124,32 +137,28 @@ object Proposer {
       epochLength: Int,
       dummyDeployOpt: Option[(PrivateKey, String)] = None
   ): Proposer[F] = {
-    // TODO: refactor proposer to get this from parent pre state
-    def getLatestSeqNumber(sender: Validator): F[Long] =
-      for {
-        dag        <- BlockDagStorage[F].getRepresentation
-        latestMsgs = dag.dagMessageState.latestMsgs
-        maxSeqNum  = latestMsgs.find(_.sender == sender).map(_.senderSeq)
-      } yield maxSeqNum.getOrElse(-1)
 
-    def createBlock(validatorIdentity: ValidatorIdentity): F[BlockCreatorResult] =
+    def createBlock(
```

Comment:
I'd rather have BlockCreator with methods to create all necessary stuff to create blocks then Proposer.